src/pymmcore_plus/mda/handlers/_ome_tiff_writer.py

```diff
@@ -0,0 +1,107 @@
+        if self._mmap is None:
+            if not self._current_sequence:
+                # just in case sequenceStarted wasn't called
+                self._set_sequence(event.sequence)  # pragma: no cover
+
+            if not (seq := self._current_sequence):
+                raise NotImplementedError(
+                    "Writing zarr without a MDASequence not yet implemented"
+                )
+
+            mmap = self._create_seq_memmap(frame, seq, meta)
+        else:
+            mmap = self._mmap
+
+        # WRITE DATA TO DISK
+        index = tuple(event.index.get(k) for k in self._used_axes)
+        mmap[index] = frame
+        mmap.flush()
+
+    # -------------------- private --------------------
+
+    def _set_sequence(self, seq: useq.MDASequence | None) -> None:
+        """Set the current sequence, and update the used axes."""
+        self._current_sequence = seq
+        if seq:
+            self._used_axes = tuple(seq.used_axes)
+
+    def _create_seq_memmap(
+        self, frame: np.ndarray, seq: useq.MDASequence, meta: dict
+    ) -> np.memmap:
+        from tifffile import imwrite, memmap
+
+        shape = (
+            *tuple(v for k, v in seq.sizes.items() if k in self._used_axes),
+            *frame.shape,
+        )
+        axes = (*self._used_axes, "y", "x")
+        dtype = frame.dtype
+
+        # see tifffile.tiffile for more metadata options
+        metadata: dict[str, Any] = {"axes": "".join(axes).upper()}
+        if seq:
+            if seq.time_plan and hasattr(seq.time_plan, "interval"):
+                interval = seq.time_plan.interval
+                if isinstance(interval, timedelta):
+                    interval = interval.total_seconds()
+                metadata["TimeIncrement"] = interval
+                metadata["TimeIncrementUnit"] = "s"
+            if seq.z_plan and hasattr(seq.z_plan, "step"):
+                metadata["PhysicalSizeZ"] = seq.z_plan.step
+                metadata["PhysicalSizeZUnit"] = "µm"
+            if seq.channels:
+                metadata["Channel"] = {"Name": [c.config for c in seq.channels]}
+        if acq_date := meta.get("Time"):
+            metadata["AcquisitionDate"] = acq_date
+        if pix := meta.get("PixelSizeUm"):
+            metadata["PhysicalSizeX"] = pix
+            metadata["PhysicalSizeY"] = pix
+            metadata["PhysicalSizeXUnit"] = "µm"
+            metadata["PhysicalSizeYUnit"] = "µm"
+
+        # TODO:
+        # there's a lot we could still capture, but it comes off the microscope
+        # over the course of the acquisition (such as stage positions, exposure times)
+        # ... one option is to accumulate these things and then use `tifffile.comment`
+        # to update the total metadata in sequenceFinished
+        imwrite(self._filename, shape=shape, dtype=dtype, metadata=metadata)
+
+        # memory map numpy array to data in OME-TIFF file
+        self._mmap = memmap(self._filename)
```

Comment:
I'm playing around with this and ran into a small problem. If I have an acquisition with one axes with length 1, the memmap call swallows that axis.
So this MDASequence for example:
``` python
2023-11-10 10:19:42,768 - pymmcore-plus - INFO - (_runner.py:311) MDA Finished: metadata={'pymmcore_widgets': {'version': '0.4.2.dev58+g3f1e2ea.d20231109'}}
axis_order=('t', 'c') 
channels=(Channel(config='488', group='Channel', exposure=100.0, do_stack=True, z_offset=0.0, acquire_every=1, camera=None),)
time_plan=TIntervalLoops(prioritize_duration=False, interval=datetime.timedelta(microseconds=200000), loops=20)
```
frameReady receives the index for channel however, leading for too many indices for the mmep. I have to do:
``` python
        self._mmap = memmap(self._filename)
        self._mmap = cast("np.memmap", self._mmap)
        self._mmap = self._mmap.reshape(shape)
```

Reply:
thanks @wl-stepp, this comment was very helpful.  using `mmap.shape = shape` immediately after instantiating the memmap appears to work to reinforce singleton dimensions with the tifffile-created memmap.  Fixed and test added.